test/integration/errors_test.go

```diff
@@ -148,3 +148,37 @@ func TestAccountEmailError(t *testing.T) {
 		})
 	}
 }
+
+func TestRejectedIdentifier(t *testing.T) {
+	t.Parallel()
+
+	// When a single malformed name is provided, we correctly reject it.
+	domains := []string{
+		"яџ–Х6яяdь}",
+	}
+
+	_, err := authAndIssue(nil, nil, domains, true)
+	test.AssertError(t, err, "authAndIssue failed")
+
+	var prob acme.Problem
+	test.AssertErrorWraps(t, err, &prob)
+	t.Logf("%#v", prob)
+	test.AssertEquals(t, prob.Type, "urn:ietf:params:acme:error:rejectedIdentifier")
+	test.AssertContains(t, prob.Detail, "Domain name contains an invalid character")
+
+	// When multiple malformed names are provided, we correctly reject all of
+	// them and reflect this in suberrors.
+	domains = []string{
+		"o-",
+		"ш№Ў",
+		"р±y",
+		"яџ–Х6яя",
+		"яџ–Х6яя`ь",
+	}
+	_, err = authAndIssue(nil, nil, domains, true)
+	test.AssertError(t, err, "authAndIssue failed")
+	test.AssertErrorWraps(t, err, &prob)
+	test.AssertEquals(t, prob.Type, "urn:ietf:params:acme:error:rejectedIdentifier")
+	test.AssertContains(t, prob.Detail, "Domain name contains an invalid character")
+	test.AssertContains(t, prob.Detail, "and 4 more problems")
+}
```

Comment:
Another possible test case.

```suggestion
	// Test with mixed valid and invalid domain names.
	domains = []string{"example.com", "р±y", "example.org"}
	_, err = authAndIssue(nil, nil, domains, true)
	test.AssertError(t, err, "Mixed valid and invalid domains: authAndIssue failed")
	test.AssertErrorWraps(t, err, &prob)
	test.AssertEquals(t, prob.Type, "urn:ietf:params:acme:error:rejectedIdentifier")
	test.AssertContains(t, prob.Detail, "Domain name contains an invalid character")
}
```

Reply:
It doesn't provide any additional coverage for the thing we care about here, which is "we're capable of returning this error both in the single-error format and the multiple-suberrors format".